app/uk/gov/hmrc/agentfirelationship/controllers/RelationshipController.scala

```diff
@@ -56,58 +56,77 @@ class RelationshipController @Inject()(
             Future successful NotFound
           } else {
-            if (appConfig.checkCesaRelationshipFlag) {
-              checkCesaService
-                .lookupCesaForOldRelationship(Arn(arn), Nino(clientId))
-                .flatMap { saAgentRefs =>
-                  if (saAgentRefs.isEmpty) {
-                    Future successful NotFound
-                  } else {
-                    if (appConfig.copyCesaRelationshipFlag) {
-                      val activeRelationship = Relationship(
-                        arn = Arn(arn),
-                        service = service,
-                        clientId = clientId,
-                        relationshipStatus = Some(RelationshipStatus.Active),
-                        startDate = LocalDateTime.now(ZoneId.of("UTC")),
-                        endDate = None,
-                        fromCesa = Some(true)
-                      )
-                      mongoService
-                        .createRelationship(activeRelationship)
-                        .flatMap(_ => mongoService.findRelationships(arn, service, clientId, RelationshipStatus.Active))
-                        .map(newResult => {
-                          auditData.set("agentReferenceNumber", arn)
-                          auditData.set("service", "personal-income-record")
-                          auditService.sendCreateRelationshipFromExisting(auditData)
-                          Ok(toJson(newResult))
-                        })
-                        .recover {
-                          case ex =>
-                            logger.error("Relationship creation failed", ex)
-                            Ok(toJson(result))
-                        }
-                    } else {
-                      Future successful Ok(toJson(result))
-                    }
-                  }
-                }
+            handleCesaRelationship(arn, service, clientId, activeRelationships)
+          }
+        }
+      }
+    }
+  }
+
+  private def handleCesaRelationship(arn: String, service: String, clientId: String, activeRelationships: List[Relationship])(
+    implicit auditData: AuditData,
+    hc: HeaderCarrier,
+    request: Request[Any]): Future[Result] =
+    if (appConfig.checkCesaRelationshipFlag) {
+      checkCesaService
+        .lookupCesaForOldRelationship(Arn(arn), Nino(clientId))
+        .flatMap { saAgentRefs =>
+          if (saAgentRefs.isEmpty) {
+            Future successful NotFound
+          } else {
+            if (appConfig.copyCesaRelationshipFlag) {
+              mongoService.findActiveClientRelationships(clientId).flatMap { activeRelationshipsAcrossAgents =>
+                if (activeRelationshipsAcrossAgents.isEmpty) copyCesaRelationship(arn, service, clientId, activeRelationships)
+                else Future successful NotFound
+              }
             } else {
-              Future.successful(NotFound)
+              Future successful Ok(toJson(activeRelationships))
```

Comment:
here we are granting access on the strength of the legacy relationship which maybe okay but we should first check if the client is authorised with another agent in MTD i.e. we should move the mongoService.findActiveClientRelationships... to above L89?

Reply:
Updated PR to move up the call to `mongoService.findActiveClientRelationships()` @geoffreywatson